### DISCUSSION TITLE
Finalizing grant application form structure

### DISCUSSION THREAD

**Discussion by chibie** [gitcoinco/grants-stack#172] [Q&A] [answered]
@schultztimothy:

👋 Aditya just reached out asking for the final schema for the application. Just thought I'd confirm here with everyone. It would be https://discord.com/channels/@me/997206435123429528/997450469213478963 with the additional fields you called out?
So

```
{
  round: String,
  recipient: String,
  project: {
    lastUpdated: blockNumber,
    title: String,
    description: String,
    website: String,
    bannerImg: String
    logoImg: String,
    metaPtr: {
      protocol: Number,
      pointer: String,
    },
    id: Number,
  },
  answers: [
    {
      question: String,
      answer: String | Number,
    },
  ],
}
```

**Comment by chibie** [comment]
1. should we include question id in the answers object? 

```
  answers: [
    {
      questionId: Number,
      question: String,
      answer: String | Number,
    },
  ],
  ```
 
2. can we set `lastUpdated` to the epoch timestamp in milliseconds instead of blockNumber? since it'll be displayed in UI
3. based on the discussion here https://github.com/gitcoinco/grants-round/discussions/171, the project id will be a js `String`

**Reply by thelostone-mc** [reply]
I would agree with all these

**Reply by tim-schultz** [reply]
They all sound good to me as well

**Comment by chibie** [comment, marked as answer]
this is the type i defined

```
export interface GrantApplication {
  /**
   * The round contract address applied to 
   */
  round: string;
  /** 
   * Recipient wallet address of grantee
   */
  recipient: string;
  /**
   * Project information
   */
  project: {
    lastUpdated: Number, // unix timestamp in milliseconds
    id: string,
    title: string,
    description: string,
    website: string,
    bannerImg?: string,
    logoImg: string,
    metaPtr: MetadataPointer
  };
  /** List of answers to questions */
  answers: Array<{
    questionId: Number,
    question: string,
    answer: string
  }>
}
```

**Reply by thelostone-mc** [reply]
Perfect. Do you have an IPFS hash i could wire into our scripts to automate applications with this structure @chibie ?

**Comment by vacekj** [comment]
Closing as stale.
